### DISCUSSION TITLE
Some info for ETL - OMOP Measurement column value_as_concept_id - follow up from Bridge2AI Office Hours 08-15-2024

### DISCUSSION THREAD

**Discussion by heschmidt04** [chorus-ai/StandardsModule#98] [Structured EHR and ETL Questions] [answered]
Jared covered the Delphi mapping in 2 different office hours so context, sql and recordings for those is on [8/15/24 here](https://drive.google.com/drive/folders/1Tk61e-0UxOQYL5gCsjPqEimqLx5hQedY) and [two previous sessions in the office hours folder here. ](https://drive.google.com/drive/folders/1vaV15KThS0c4ioo0vEJCvUJ1aRmKI55b)

What came up when folks were looking for example Sputnum measurement of "Thick" with concept id 4081323 into OMOP Measurement value_as_concept_id...
It looked like that there may be more in a neighboring concept class under the "meas value" domain. 
[Example search here at athena.ohdsi.org](https://athena.ohdsi.org/search-terms/terms?domain=Meas+Value&conceptClass=Answer&conceptClass=Qualifier+Value&page=1&pageSize=500&query=thick&boosts)

For those who have existing OMOP set up and don't have access to their sites Athena Vocabulary tables to change them for Bridge2AI vocabulary 2million+ ids, or are using OMOP source_to_concept_map then ...

The SQL below in some form below may be handy for use in determining what Athena concept ids in the "meas value" domain in the Concept table should be used as a "number" for a measurement that is a string to populate OMOP Measurement table column value_as_concept_id. 

The measurement value that is a string may also need to be unpacked (because it has ; or , or () or - , or a mix of them in the value and indicates that it is many measurements in one single record) 

In Snowflake I can do that split inline on specific delimiters with the function LATERAL.

Previous to that inline function,
I use the same CASE logic for [OMOP CDM observation table](https://ohdsi.github.io/CommonDataModel/cdm54.html#observation) to break out the OMOP measurement table with value_as_string and value_as_number so that when I do the remaining work to filter for measurements that have strings as values - it is a smaller data set to do work on. 

```
    CASE
        WHEN TRY_TO_NUMBER(value_as_number) IS NOT NULL THEN TRY_TO_NUMBER(value_as_number)
        ELSE NULL
    END AS VALUE_AS_NUMBER,
    CASE
        WHEN TRY_TO_NUMBER(value_as_number) IS NULL THEN LEFT(value_as_number,50)
        ELSE NULL
    END AS VALUE_AS_STRING,
```

Overall - to map all measurement strings to athena concept ids in column value_as_concept_id.
This may be helpful to create a smaller dimension table with which to join to the ETL tables by product tables that your site creates to populate the [OMOP CDM measurement table.](https://ohdsi.github.io/CommonDataModel/cdm54.html#measurement) 

```
CREATE TABLE stcm_meas_value_mappings AS 
WITH athena_meas_value_mappings AS (
SELECT 
      concept_id AS meas_value_as_concept_id, 
      lower(concept_name) AS concept_name,
      lower(domain_id) AS domain_id,
      vocabulary_id,
      concept_class_id,
      concept_code, 
      valid_start_date,
      valid_end_date,
      invalid_reason
FROM  concept
WHERE domain_id IN ('observation','meas value' ) 
AND   concept_class_id in ('Answer','Qualifier Value')
AND   standard_concept = 'S' 
-- I found filtering on vocab id can limit the ability to pick from available matches. YMMV.
-- AND vocabulary_id = 'SNOMED' 
)
SELECT * FROM athena_meas_value_mappings;
```

**Comment by jshoughtaling** [comment, marked as answer]
@heschmidt04 - thanks for starting this thread! Just one minor addition: any value mappings to standard concepts outside of the MEAS VALUE domain *cannot* be used in the `value_as_concept_id` field of the MEASUREMENT table. That field is restricted to standard MEAS VALUE concepts. 

For those flowsheet descriptions that map to an OBSERVATION concept, you can map their associated categorical values to any domain you see fit ([see example here](https://github.com/chorus-ai/StandardsModule/discussions/99#discussioncomment-10394877))
